File: maths/matmul/matmul.py
import sys
import numpy as np
import time

n = int(sys.argv[1])

start = time.time()
A = np.random.uniform(0,1,n*n).reshape((n,n))
B = np.random.uniform(0,1,n*n).reshape((n,n))
C = np.zeros((n,n))
stop = time.time()
print("time: ", stop-start)

start = time.time()
C = A @ B
stop = time.time()
print("time: ", stop-start)

#print(A)
#print(B)
#print(C)

File: maths/matmul/matmul.ml
#! /usr/bin/env ocaml

let rec getFirst n = match n with
	| [] -> failwith "empty"
	| [e] -> e
	| e :: _ :: r -> getFirst [e] ;;


print_string "Hello World !\n";;
Random.self_init();;
let j = Random.int 15 ;;
Printf.printf ("%i\n") j ;;

let s1 = Random.get_state ();;
let n = Pervasives.int_of_string Sys.argv.(1) ;;
let ccols = Array.make n 0.00 ;;
let c = Array.make n ccols;;
let bcols = Array.make n (Random.State.int s1 15) ;;
let b = Array.make n bcols;;
let acols = Array.make n (Random.int 15) ;;
let a = Array.make n acols;;

for i = 0 to n-1 do
	for j = 0 to n-1 do
		Printf.printf "%i " a.(i).(j) 
	done;
	Printf.printf "\n"
done;;


